apps/src/lab2/Lab2Registry.ts

```diff
@@ -1,13 +1,17 @@
 // Registry for Lab singletons
 
+import LabMetricsReporter from './Lab2MetricsReporter';
 import ProjectManager from './projects/ProjectManager';
 
 export default class Lab2Registry {
-  projectManager: ProjectManager | null;
+  private projectManager: ProjectManager | null;
+  private metricsReporter: LabMetricsReporter;
+
   private static _instance: Lab2Registry;
 
   constructor() {
     this.projectManager = null;
+    this.metricsReporter = new LabMetricsReporter();
   }
 
   public static getInstance(): Lab2Registry {
```

Comment:
I like the mods, except for this method which feels excessively wordy. Since we're _always_ (I assume?) calling `getInstance()`.

Could this be wrapped up so we can just import the `Lab2Registry` singleton directly?

Alternatively - could we wrap up and vend out individual reporters? So we can just import a `MetricsReporter` instead of getting the registry + an instance + the metrics reporter (which is still a singleton since its hanging off of the Lab2Registry singleton _instance, right?)

Was there another reason to access the reporter in this manner?

Reply:
Yeah, I suppose we could export a singleton `Lab2Registry` instead of calling `getInstance()`? Agree that it would be less wordy, though I'd probably want to do that as a separate change since it would affect how we access other fields in this singleton (like `ProjectManager`). I do want to keep a single metrics reporter (rather than vending out individual reporter) because it ensures that all lab2 labs in the same session are using the same reporter instance. The same reporter instance gets automatically updated with common fields whenever those change.